### ISSUE TITLE
Glide on Windows writes the lock file with incorrect slashes

### ISSUE THREAD

**Issue by colemickens** [Masterminds/glide#235]
Deduplication and identification of included packages seems to be affected as well.

Here's the diff of what glide did when running `glide up` on windows:

```diff
diff --git a/glide.lock b/glide.lock
index 03c385f..a14e606 100644
--- a/glide.lock
+++ b/glide.lock
@@ -1,19 +1,47 @@
 hash: 46cd6d2d3f2840288fad453b3d7d8309d53ed56f84794a6edba7c71968b95142
-updated: 2016-01-28T01:15:34.905936244-08:00
+updated: 2016-01-29T12:20:43.7050665-08:00
 imports:
 - name: github.com/Azure/go-autorest
-  version: 20a00a4f1f2a2abdfb98e6beb1126f1d36d7063a
+  version: 8ced4882956df3128b6d81a183d8e0757ac825b3
   subpackages:
   - /autorest
   - autorest/azure
   - autorest/date
   - autorest/to
 - name: github.com/dgrijalva/jwt-go
-  version: f164e17f59b82642a3895ba065c385db6c547344
+  version: afef698c326bfd906b11659432544e5aae441d44
+- name: github.com\Azure\go-autorest
+  version: 8ced4882956df3128b6d81a183d8e0757ac825b3
+- name: github.com\dgrijalva\jwt-go
+  version: afef698c326bfd906b11659432544e5aae441d44
 - name: golang.org/x/crypto
-  version: 346896d57731cb5670b36c6178fc5519f3225980
+  version: 1f22c0103821b9390939b6776727195525381532
   subpackages:
   - /pkcs12
+- name: golang.org\x\crypto
+  version: 1f22c0103821b9390939b6776727195525381532
+- name: golang.org\x\crypto\blowfish
+  version: ""
+- name: golang.org\x\crypto\nacl\secretbox
+  version: ""
+- name: golang.org\x\crypto\openpgp\armor
+  version: ""
+- name: golang.org\x\crypto\openpgp\errors
+  version: ""
+- name: golang.org\x\crypto\openpgp\packet
+  version: ""
+- name: golang.org\x\crypto\openpgp\s2k
+  version: ""
+- name: golang.org\x\crypto\pkcs12
+  version: ""
+- name: golang.org\x\crypto\pkcs12\internal\rc2
+  version: ""
+- name: golang.org\x\crypto\poly1305
+  version: ""
+- name: golang.org\x\crypto\salsa20\salsa
+  version: ""
 - name: gopkg.in/check.v1
-  version: 11d3bc7aa68e238947792f30573146a3231fc0f1
+  version: 4f90aeace3a26ad7021961c297b22c42160c7b25
+- name: gopkg.in\check.v1
+  version: 4f90aeace3a26ad7021961c297b22c42160c7b25
 devImports: []
```

Note that go-autorest is re-added to the lock file, even though it's already there.

Here's the diff it should have created: (this was generated from `glide up` on a Linux box)
```diff
diff --git a/glide.lock b/glide.lock
index 58c0c35..03c385f 100644
--- a/glide.lock
+++ b/glide.lock
@@ -1,21 +1,19 @@
 hash: 46cd6d2d3f2840288fad453b3d7d8309d53ed56f84794a6edba7c71968b95142
-updated: 2016-01-29T13:16:26.939463551-08:00
+updated: 2016-01-28T01:15:34.905936244-08:00
 imports:
 - name: github.com/Azure/go-autorest
-  version: 8ced4882956df3128b6d81a183d8e0757ac825b3
+  version: 20a00a4f1f2a2abdfb98e6beb1126f1d36d7063a
   subpackages:
   - /autorest
   - autorest/azure
   - autorest/date
   - autorest/to
 - name: github.com/dgrijalva/jwt-go
-  version: afef698c326bfd906b11659432544e5aae441d44
-  subpackages:
-  - .
+  version: f164e17f59b82642a3895ba065c385db6c547344
 - name: golang.org/x/crypto
-  version: 1f22c0103821b9390939b6776727195525381532
+  version: 346896d57731cb5670b36c6178fc5519f3225980
   subpackages:
   - /pkcs12
 - name: gopkg.in/check.v1
-  version: 4f90aeace3a26ad7021961c297b22c42160c7b25
+  version: 11d3bc7aa68e238947792f30573146a3231fc0f1
 devImports: []
```

**Comment by mattfarina** [comment]
Thanks for providing this bug. I know what's going on. Working with the filesystem in a cross platform manner can cause bugs like this to pop up. I'll fix it shortly.

**Comment by mattfarina** [comment]
@colemickens Can you check the latest on master. I think I fixed this.

**Comment by colemickens** [comment]
@garimakhulbe Can you try this on your Windows machine and report back the diff between glide.lock before and after running `glide up` ?

**Comment by garimakhulbe** [comment]
Below is the glide up output with glide.lock diff.

![image](https://cloud.githubusercontent.com/assets/9362976/12734915/79d2f718-c8f8-11e5-9ab3-f769054b0697.png)


![image](https://cloud.githubusercontent.com/assets/9362976/12734929/97d1904e-c8f8-11e5-9ab8-3cf660cf7b37.png)

**Comment by colemickens** [comment]
@mattfarina We confirmed that we were on the very latest of glide when that was generated. It seems to not be resolved.

**Comment by mattfarina** [comment]
Thanks for your patience. I'm now setup to work on this in my Windows environment and I'll test it against the Azure SDK for Go.

**Comment by mattfarina** [comment]
@colemickens @garimakhulbe can you take a look at the latest on master. I have it working in my Windows environment now.

**Comment by colemickens** [comment]
CC @boumenot

**Comment by garimakhulbe** [comment]
After fetching the latest glide repo. 

![image](https://cloud.githubusercontent.com/assets/9362976/12761346/7752c322-c99f-11e5-9317-44574ec67dbd.png)

**Comment by colemickens** [comment]
@mattfarina It looks like there might be one remaining issue, since it's trying to add the unused subpackages in Windows still.

**Comment by mattfarina** [comment]
@colemickens I'll take a look at that.

**Comment by mattfarina** [comment]
The latest on master is not venerating the right subpackages for me. Can you try again.

**Comment by colemickens** [comment]
@mattfarina Sorry, I don't follow, are you saying there's a problem with `azure-sdk-for-go` master?

**Comment by mattfarina** [comment]
@colemickens Well, that typo of mine messes with the meaning. Sorry about that. The latest on master is **_now generating_** the right subpackages for me. Can you please try again?

**Comment by colemickens** [comment]
@garimakhulbe Can you give it a shot again? I still don't have a Windows machine setup with the relevant stuff.

**Comment by garimakhulbe** [comment]
![image](https://cloud.githubusercontent.com/assets/9362976/12902914/d8bc1982-ce78-11e5-96f7-172a197a59d4.png)

Ran with all latest.

**Comment by colemickens** [comment]
I still see two issues:

* To my knowledge, `rc2` is still being improperly added. `azure-sdk-for-go` does not directly use the `rc2` package. (Actually, I think even the references to pkcs12 have been removed, I should remove it entirely)

* `/autorest` and `autorest` are being treated as different subpackages.

**Comment by mattfarina** [comment]
This latest commit should fix the issue where the same package is present with and without the leading /.

The `pkcs12` package imports the `pkcs12/internal/rc2` package which is why it shows up in the lock file. It's imported in the `pkcs12/crypto.go` file. This is the complete tree of referenced packages which is why it's present in the lock.

Is there anything else here that needs fixing?

**Comment by garimakhulbe** [comment]
![image](https://cloud.githubusercontent.com/assets/9362976/12938929/a3451a1e-cf6e-11e5-93fe-91764bfa04f2.png)

LGTM. @colemickens, please confirm.

**Comment by colemickens** [comment]
I'm just not sure why rc2 was only added in Windows. Transitive
dependencies are normally shown in the lock file, not the yaml file, in my
experience.

**Comment by colemickens** [comment]
Maybe the exception is when the transitive dependency is a subpackage or
from the same repo? But it doesn't get added in Linux (that was about two
weeks ago though).

**Comment by colemickens** [comment]
We can safely ignore my first comment from a few hours ago since we are in fact looking at the lock file in the diff. 

I updated `glide` on my Linux box. It is now producing the same diff as what @garimakhulbe has shown above. 

However, `rc2` has always been required by `pkcs12`... so it looks like we were actually seeing another bug here that has been fixed in the last few weeks. Does that sound accurate @mattfarina? If so, we can certainly close this.

Thanks for being so responsive.
